### PR TITLE
Modify UpdateCommand, UpdateCommandParser

### PR DESCRIPTION
**Pull request by qlchan24**
- Add support for Supplier and all fields of product

- Change implementation to check for optional product fields

- Throw error if product exists but no fields supplied

- Add new UpdateProductDescriptor and its Builder class

### CODE REVIEW

File: src/main/java/seedu/clinic/logic/commands/UpdateCommand.java [hunk 1]
```diff
@@ -1,74 +1,121 @@
...
 import seedu.clinic.model.Model;
 import seedu.clinic.model.attribute.Name;
+import seedu.clinic.model.attribute.Tag;
 import seedu.clinic.model.product.Product;
+import seedu.clinic.model.supplier.Supplier;
 import seedu.clinic.model.warehouse.Warehouse;
 
 /**
- * Update the stock for a product in a specific warehouse.
+ * Update the quantity/tags for a product in a specific warehouse/supplier, or adds the product if it does not
```
Comment (jeffreytjs) [inline src/main/java/seedu/clinic/logic/commands/UpdateCommand.java]: Should be Updates I think

**Review by jeffreytjs — COMMENTED** [review]
LGTM!

File: src/main/java/seedu/clinic/logic/commands/UpdateCommand.java (hunk 1 above)
Comment (qlchan24) [reply src/main/java/seedu/clinic/logic/commands/UpdateCommand.java]: oh yes thanks!

**Review by Criss-Wang — APPROVED** [review]
LGTM!

**Pull request merged**